app/models/shelter.rb

```diff
@@ -1,3 +1,3 @@
 class Shelter < ApplicationRecord
-  has_many :pets
+  has_many :pets, dependent: :destroy
```

Comment:
I added `dependent: :destroy` so that if a shelter gets deleted, so do its pets.